src/tests.rs

```diff
@@ -0,0 +1,203 @@
+#![cfg(test)]
+
+use super::mock::{
+    new_test_ext, Balances, Bridge, Origin, System, Test, ENDOWED_BALANCE, ENDOWED_ID, USER,
+    VALIDATOR_A, VALIDATOR_B, VALIDATOR_C,
+};
+use super::*;
+use frame_support::{assert_noop, assert_ok};
+use frame_system::{EventRecord, Phase};
+
+use sp_core::{blake2_256, H256};
+use sp_runtime::testing::TestValidity::Valid;
+
+#[test]
+fn set_get_address() {
+    new_test_ext(1).execute_with(|| {
+        assert_ok!(Bridge::set_address(
+            Origin::signed(VALIDATOR_A),
+            vec![1, 2, 3, 4]
+        ));
+        assert_eq!(<EmitterAddress>::get(), vec![1, 2, 3, 4])
+    })
+}
+
+#[test]
+fn asset_transfer_success() {
+    new_test_ext(1).execute_with(|| {
+        let chain_id = vec![1];
+        let to = vec![2];
+        let token_id = vec![3];
+        let metadata = vec![];
+
+        assert_ok!(Bridge::whitelist_chain(
+            Origin::signed(VALIDATOR_B),
+            chain_id.clone()
+        ));
+        assert_ok!(Bridge::receive_asset(
+            Origin::ROOT,
+            chain_id.clone(),
+            to.clone(),
+            token_id.clone(),
+            metadata.clone()
+        ));
```

Comment:
Please also test whether events are dispatched properly (in this and other tests), see for an example: https://github.com/centrifuge/substrate-pallet-multi-account/blob/master/src/lib.rs#L888

Reply:
Really useful snippet. I was previously using `expect_event` but removed it when we started having multiple events fire in a single call. Will revisit.